app/services/post_persister_task_service.rb

```diff
@@ -13,7 +13,19 @@ def validate_options
 
   def update_source
     source = Source.find(@options[:task].source_id)
-    source.update!(:last_successful_refresh_at => @options[:task][:input]["refresh_request_at"]) if @options[:task].status == "ok"
-    source.update!(:refresh_finished_at => Time.current)
+    @options[:task].status == "ok" ? source.update!(ok_options) : source.update!(error_options)
+
+    Rails.logger.info("Source #{source.id}: refresh finished at #{source.refresh_finished_at}, state: #{source.refresh_state}")
+  end
+
+  def ok_options
+    {:last_successful_refresh_at => @options[:task][:input]["refresh_request_at"],
+     :refresh_finished_at        => Time.current,
+     :refresh_state              => "Done"}
+  end
+
+  def error_options
```

Comment:
@hsong-rh This logic is also needed when the Upload task has errors. It's possible that the tower token has expired on the client and it can't talk to the Ansible Tower. Which also begs a question if we need another field in the source object called "last_refresh_message" where we can store the error.